src/components/reports/ReportForm.tsx

```diff
@@ -0,0 +1,208 @@
+import React, {useState} from "react";
+import {Form, Field, Formik, ErrorMessage} from 'formik';
+import * as Yup from "yup"
+
+import Card from "../UI/Card";
+import classes from "./ReportForm.module.css";
+import {addressSchema, patientSchema, reportDetailSchema, sampleSchema, variantSchema} from "./formDataValidation";
+
+const FormValidation = Yup.object().shape({
+  address: addressSchema,
+  patient: patientSchema,
+  sample: sampleSchema,
+  variant: variantSchema,
+  result: reportDetailSchema,
+}).required();
```

Comment:
I've not used Yup before. But it looks decent!

Reply:
yeah I quite like how you can generate the type definition from it!